be/src/storage/tablet_updates.cpp

```diff
@@ -3256,16 +3256,17 @@ Status TabletUpdates::load_snapshot(const SnapshotMeta& snapshot_meta, bool rest
         l2.unlock();                     // _rowsets_lock
         _try_commit_pendings_unlocked(); // may acquire |_rowset_stats_lock| and |_rowsets_lock|
 
+        _update_total_stats(_edit_version_infos[_apply_version_idx]->rowsets, nullptr, nullptr);
+        _apply_version_changed.notify_all();
+        l1.unlock();
```

Comment:
I will be better to add some comments in the code base to explain why the lock is unlock here.

Reply:
> I will be better to add some comments in the code base to explain why the lock is unlock here.

done, add some comments in https://github.com/StarRocks/starrocks/pull/18488/commits/6c71ec140154bfcb073ec4f08489c78dcf05108b